File: src/main/java/com/github/squirrelgrip/meter/database/DatabaseSession.java
package com.github.squirrelgrip.meter.database;

import com.github.squirrelgrip.meter.domain.MeterReading;

public interface DatabaseSession extends AutoCloseable {
    void start();
    void commit();
    void rollback();

    void save(MeterReading meterReading);
    Long count();
}

File: src/main/kotlin/com/github/squirrelgrip/meter/database/DatabaseSession.kt
package com.github.squirrelgrip.meter.database

import com.github.squirrelgrip.meter.entity.MeterReading
import java.lang.AutoCloseable

interface DatabaseSession : AutoCloseable {
    fun start()
    fun commit()
    fun rollback()

    fun save(meterReading: MeterReading?)
    fun count(): Long

    val isTransactionActive: Boolean
}
